InfiniTAM/Engine/RosPoseSourceEngine.cpp

```diff
@@ -15,146 +13,100 @@ namespace Engine {
-             .getY();
+        tf_world_to_camera_transform_current_.getOrigin().getY();
     infinitam_translation_vector_.z =
-        -tf_infinitam_origin_to_camera_current_translation_in_infinitam_frame_rotated
-             .getZ();
+        tf_world_to_camera_transform_current_.getOrigin().getZ();
+
+    // Invert the rotation since we use this in the perspective of the camera.
+    tf_world_to_camera_transform_current_.setBasis(
+        tf_world_to_camera_transform_current_.getBasis().inverse());
 
     infinitam_rotation_matrix_.m00 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(0)
-            .getX();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(0).getX();
     infinitam_rotation_matrix_.m10 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(0)
-            .getY();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(0).getY();
     infinitam_rotation_matrix_.m20 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(0)
-            .getZ();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(0).getZ();
     infinitam_rotation_matrix_.m01 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(1)
-            .getX();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(1).getX();
     infinitam_rotation_matrix_.m11 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(1)
-            .getY();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(1).getY();
     infinitam_rotation_matrix_.m21 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(1)
-            .getZ();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(1).getZ();
     infinitam_rotation_matrix_.m02 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(2)
-            .getX();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(2).getX();
     infinitam_rotation_matrix_.m12 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(2)
-            .getY();
+        tf_world_to_camera_transform_current_.getBasis().getColumn(2).getY();
     infinitam_rotation_matrix_.m22 =
-        tf_infinitam_origin_to_camera_transform_relative_.getBasis()
-            .getColumn(2)
-            .getZ();
-
-    ROS_DEBUG_STREAM("\n transformation infinitam origin to camera pose: \n"
-                     << infinitam_translation_vector_ << "\n");
-    ROS_DEBUG_STREAM("\n transformation infinitam origin to camera pose: \n"
-                     << infinitam_rotation_matrix_ << "\n");
+        tf_world_to_camera_transform_current_.getBasis().getColumn(2).getZ();
```

Comment:
There is not method to make this cleaner... the infinitam_rotation_matrix type is a custom type from infinitam which does not have other assign methods.